Descent3/CMakeLists.txt

```diff
@@ -299,6 +299,8 @@ add_executable(Descent3
   ${HEADERS} ${CPPS} ${PLATFORM_CPPS} ${INCS}
 )
 target_link_libraries(Descent3
+  $<TARGET_NAME_IF_EXISTS:SDL2::SDL2main>
+  $<IF:$<TARGET_EXISTS:SDL2::SDL2>,SDL2::SDL2,SDL2::SDL2-static>
```

Comment:
trying to understand this line. Can't we assume we always have the SDL2::SDL2 target ? How would the find package only find the static build ?

Reply:
If using the vcpkg target, there is always either a `SDL2::SDL2` target for the dynamic version, or a `SDL2::SDL2-static` one. The `find_package` will find the associated CMake package, but a package can have multiple targets, so this line basically just tries to find first the dynamic one using generator expressions, then the static one.

Reply:
got it